### PR TITLE
island-ui / Hyphen

### PR DESCRIPTION
**Pull request by barabrian**
# Hyphen

## What

Add soft hyphens to icelandic text

## Screenshots / Gifs

long-text-hypen

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/island-ui/core/src/lib/Hyphen/Hyphen.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  minRight?: number
+  minLeft?: number
+}
+
+export const Hyphen = ({
+  children,
+  minRight = 4,
+  minLeft = 4,
+}: HyphenProps) => {
+  const text = useMemo(() => hypenateText(children, { minRight, minLeft }), [
```
Comment (DavidBachmann) [inline libs/island-ui/core/src/lib/Hyphen/Hyphen.tsx:38]: 🥇

File: libs/island-ui/core/src/lib/Hyphen/Hyphen.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
+import React, { useMemo } from 'react'
+import ReactHtmlParser from 'react-html-parser'
+import Hypher from 'hypher'
+import { is } from './patterns'
```
Comment (DavidBachmann) [inline libs/island-ui/core/src/lib/Hyphen/Hyphen.tsx:4]: I wonder if we can/should dynamically import this

**Review by DavidBachmann — DISMISSED** [review]
I like it

File: libs/island-ui/core/src/lib/Hyphen/Hyphen.tsx (hunk 2 above)
Comment (barabrian) [reply libs/island-ui/core/src/lib/Hyphen/Hyphen.tsx:4]: That might be a good idea, I think this is sufficient while we only have one pattern but we should look into having this dynamically imported when/if we add locale support

**Review by DavidBachmann — APPROVED** [review]

**Review by sindrig — APPROVED** [review]
LGTM

**Pull request merged**